uvicorn/middleware/proxy_headers.py

```diff
@@ -14,11 +14,21 @@ class ProxyHeadersMiddleware:
     def __init__(self, app, trusted_hosts="127.0.0.1"):
         self.app = app
         if isinstance(trusted_hosts, str):
-            self.trusted_hosts = [item.strip() for item in trusted_hosts.split(",")]
+            self.trusted_hosts = {item.strip() for item in trusted_hosts.split(",")}
         else:
-            self.trusted_hosts = trusted_hosts
+            self.trusted_hosts = set(trusted_hosts)
         self.always_trust = "*" in self.trusted_hosts
 
+    def get_trusted_client_host(
+        self, x_forwarded_for_hosts
+    ):  # type: (List[str]) -> str
+        if self.always_trust:
+            return x_forwarded_for_hosts[0]
+
+        for host in reversed(x_forwarded_for_hosts):
```

Comment:
Hi @b0g3r,

Doesn't adding `reversed()` here negate the idea of returning the original client ip address in the x_forwarded_for list (which should be the very first ip address)?

Reply:
It's just a security feature: we should return only the last non-trusted host.

Trivial case:
 - Proxy-server A
 - Proxy-server B
 - Client C
 - We trust both of proxies (`trusted_hosts = A, B`) or trust every proxy (`trusted_hosts = *`)
So in this scenario, a header will be `X-Forwarded-For: C, B, A` and we can securely extract C as a trusted client address.

Unfortunately, IRL we have a situation when we have a chain of proxies, and we can trust only a few of them, e.g:
 - Proxy-server A
 - Proxy-server B
 - Client C
 - We trust only A (`trusted_hosts = A`)

In this case, non-trusted proxy-server B can replace the real client address, so we should return the first non-trusted host and it will be `B`. Because the chain is actually reverted, we should check it also with `reversed`.